hs-bindgen/src-internal/HsBindgen/Frontend/AST/External.hs

```diff
@@ -195,10 +196,20 @@ data Typedef = Typedef {
 
 data Function = Function {
       functionArgs :: [Type]
+    , functionAttrs :: [FunctionAttr]
     , functionRes  :: Type
     }
   deriving stock (Show, Eq, Generic)
 
+-- | Function attributes specify function properties that may help the C
```

Comment:
No need to repeat the defintion, `External` can just re-export definitions from `Internal` when they are unchanged.